### PR TITLE
Extract equal functionality for config items to macros

### PR DESCRIPTION
**Pull request by EPashkin**

### CODE REVIEW

**Comment by gkoz** [comment]
This looks like it could just be a trait, partly because newtypes don't seem to be called for.

File: src/config/identables.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                        if let Some(par) = $item::parse(par, object_name) {
+                            v.push(par);
+                        }
+                    }
+                }
+
+                $items(v)
+            }
+
+            pub fn matched(&self, name: &str) -> Vec<&$item> {
```
Comment (gkoz) [inline src/config/identables.rs]: We might try experimenting with trait objects, e.g. would `Box<Iterator<Item = &$item>>` here?

**Comment by EPashkin** [comment]
For trait case needed to many declaration:
only `matches` needed 2 function, for `parse` also 2 functions: 'parse` for item and `from` for items.
```
impl Identable for Function {
    fn ident(&self) -> &Ident { &self.ident }
}

impl Identables for Functions {
    type Item = Function;

    fn iter(&self) -> Iter<Self::Item> { self.0.iter() }
}
```

File: src/config/identables.rs (hunk 1 above)
Comment (EPashkin) [inline src/config/identables.rs]: Currently it used in `Function::matched_parameters` with `extend_from_slice`

**Comment by gkoz** [comment]
Some of boilerplate would come from the existence of newtypes but is there any point to hiding that `Vec<Function>` is a vector? A blanket implementation for `Vec<T: Parse + AsRef<Ident>>` should do the job.

**Comment by EPashkin** [comment]
You right about unneeded hiding. I add other PR with traits to compare result.

**Comment by EPashkin** [comment]
Closing in favor of #212

**Pull request closed without merging**